src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/DependenciesTreeSearchProvider.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        private readonly ImmutableArray<IDependenciesTreeSearchProvider> _providers;
+        private readonly JoinableTaskContext _joinableTaskContext;
+        private readonly IVsHierarchyItemManager _hierarchyItemManager;
+        private readonly IProjectServiceAccessor _projectServiceAccessor;
+        private readonly IRelationProvider _relationProvider;
+
+        [ImportingConstructor]
+        public DependenciesTreeSearchProvider(
+            [ImportMany] IEnumerable<IDependenciesTreeSearchProvider> providers,
+            JoinableTaskContext joinableTaskContext,
+            IVsHierarchyItemManager hierarchyItemManager,
+            IProjectServiceAccessor projectServiceAccessor,
+            IRelationProvider relationProvider)
+        {
+            _providers = providers.ToImmutableArray();
+            _joinableTaskContext = joinableTaskContext;
+            _hierarchyItemManager = hierarchyItemManager;
+            _projectServiceAccessor = projectServiceAccessor;
+            _relationProvider = relationProvider;
+        }
+
+        public void Search(IRelationshipSearchParameters parameters, Action<ISearchResult> resultAccumulator)
+        {
+            Requires.NotNull(parameters, nameof(parameters));
+            Requires.NotNull(resultAccumulator, nameof(resultAccumulator));
+
+            if (_providers.Length == 0)
+            {
+                // No providers registered
+                return;
+            }
+
+            if (!parameters.Options.SearchExternalItems)
+            {
+                // Consider the dependencies tree as containing 'external items', allowing the
+                // tree to be excluded from search results via this option.
+                return;
+            }
+
+            using var context = new DependenciesTreeSearchContext(parameters, resultAccumulator);
+
+            _joinableTaskContext.Factory.Run(SearchSolutionAsync);
+
+            Task SearchSolutionAsync()
+            {
+                // Search projects concurrently
+                return Task.WhenAll(_projectServiceAccessor.GetProjectService().LoadedUnconfiguredProjects.Select(SearchProjectAsync));
+            }
+
+            Task SearchProjectAsync(UnconfiguredProject unconfiguredProject)
+            {
+                IUnconfiguredProjectVsServices? projectVsServices = unconfiguredProject.Services.ExportProvider.GetExportedValue<IUnconfiguredProjectVsServices>();
+                IProjectTree? dependenciesNode = projectVsServices?.ProjectTree.CurrentTree?.FindChildWithFlags(DependencyTreeFlags.DependenciesRootNode);
+
+                if (projectVsServices == null || dependenciesNode == null)
+                {
+                    return Task.CompletedTask;
+                }
+
+                var targets = new List<string>();
+
+                foreach (ConfiguredProject configuredProject in projectVsServices.Project.LoadedConfiguredProjects)
```

Comment:
Why loaded projects? This property will return all projects that have ever been loaded. I think you want "implicitly active configs" via IActiveConfigurationGroupService2.